File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/http2native/Http2Response.java
package ru.kmorozov.gbd.core.logic.connectors.http2native;

import ru.kmorozov.gbd.core.logic.connectors.Response;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.http.HttpResponse;

public class Http2Response implements Response {

    private final HttpResponse response;
    private InputStream is;

    Http2Response(HttpResponse response) {
        this.response = response;

        if (response.body() instanceof byte[])
            is = new ByteArrayInputStream((byte[]) response.body());
    }

    @Override
    public InputStream getContent() throws IOException {
        return is;
    }

    @Override
    public String getImageFormat() {
        return response.headers().firstValue("content-type").get().split("/")[1];
    }

    @Override
    public void close() throws IOException {
        if (is != null)
            is.close();
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/http2native/Http2Response.kt
package ru.kmorozov.gbd.core.logic.connectors.http2native

import ru.kmorozov.gbd.core.logic.connectors.Response
import java.io.IOException
import java.io.InputStream
import java.net.http.HttpResponse

class Http2Response internal constructor(private val response: HttpResponse<*>) : Response {
    override val statusCode: Int
        get() = response.statusCode()

    override val content: InputStream
        get() = response.body() as InputStream

    override val imageFormat: String
        get() = response.headers().firstValue("content-type").get().split("/".toRegex()).dropLastWhile { it.isEmpty }.toTypedArray()[1]

    override val headers: String
        get() = response.headers().toString()

    @Throws(IOException::class)
    override fun close() {
        content.close()
    }
}
